test/RegisterSpec.hs

```diff
@@ -2,40 +2,94 @@
 
 module RegisterSpec where
 
-import Register
-import Test.Hspec
-import Test.QuickCheck
 import Bit
+import Control.Monad
 import FixedVector
 import FixedVectorSpec ()
 import GHC.TypeLits
+import Register
+import Test.Hspec
+import Test.QuickCheck
 import Prelude hiding ((+))
-import Control.Monad
 
 notChangesEachBit :: KnownNat n => Register n -> Expectation
 notChangesEachBit r = zipWithM_ shouldNotBe (toList r) (toList $ Register.not r)
 
 notIsItsOwnInverse :: KnownNat n => Register n -> Expectation
 notIsItsOwnInverse r = Register.not (Register.not r) `shouldBe` r
 
+allZeros :: KnownNat n => Register n
 allZeros = fromList [O, O ..]
+
+allOnes :: KnownNat n => Register n
 allOnes = fromList [I, I ..]
 
 testAnd :: KnownNat n => Register n -> Expectation
 testAnd r = do
     Register.and r r `shouldBe` r
-    Register.and r (Register.not r) `shouldBe` fromList [O, O ..]
-    -- TODO(thies): and with allZeros and allOnes, implement other tests
+    Register.and r (Register.not r) `shouldBe` allZeros
+    Register.and r allZeros `shouldBe` allZeros
```

Comment:
I think we can use `and` instead of `Register.and` since we're not importing the full Prelude.

Same goes for other operations such as `or`.